go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -232,43 +256,50 @@ func transformProjection(ctx *plancontext.PlanningContext, op *operators.Project
 		return useSimpleProjection(ctx, op, cols, src)
 	}
 
-	expressions := slice.Map(op.Projections, func(from operators.ProjExpr) sqlparser.Expr {
-		return from.GetExpr()
-	})
+	ap, err := op.GetAliasedProjections()
+	if err != nil {
+		return nil, err
+	}
 
-	failed := false
```

Comment:
@harshit-gangal for SELECT queries, this is done during offset planning on the operators. We should be able to do this there as well, right? Not for this PR

Reply:
Do you mean testing that ProjExpr received is either an eval or offset?
But it looks like we were only using failed to create a projection engine primitive....

Reply:
It looks like we fail now. What was working before that isn't working now?

Reply:
This logic could be moved to offset planning.